src/Generator/CodeGenerator.php

```diff
@@ -117,39 +147,129 @@ public function writeTraitForClass(ReflectionClass $class)
         }
     }
 
+    public function writeEnumeratorAccessorsForClass(ReflectionClass $class): array
+    {
+        $metadata  = $this->getMetadataForClass($class);
+        $fs        = new Filesystem();
+        $generated = [];
+
+        foreach ($metadata['properties'] as $info) {
+            /* @var $info PropertyInformation */
+            if (! $info->willGenerateEnumeratorAccessors()) {
+                continue;
+            }
+
+            foreach ($info->getEnumeratorsToGenerate() as $enumerator) {
+                $cache_id = $enumerator->getEnumeratorClass();
+                if (isset($this->enum_class_cache[$cache_id])) {
+                    // Do not generate the same class twice.
+                    continue;
+                }
+
+                $this->enum_class_cache[$cache_id] = $this->generateEnumeratorAccessors($enumerator);
+
+                $reflector   = new \ReflectionClass($enumerator->getEnumeratorClass());
+                $generated[] = $enumerator->getEnumeratorClass();
+                $path        = dirname($reflector->getFileName()) . DIRECTORY_SEPARATOR . $this->namespace;
+                $filename    = $path
+                    . DIRECTORY_SEPARATOR
+                    . $reflector->getShortName()
+                    . $this->enum_name_suffix
+                    . '.php';
+
+                $this->validateEnumEntity($info->getType());
+
+                $fs->mkdir($path);
+                $fs->dumpFile($filename, $this->enum_class_cache[$cache_id]);
+            }
+        }
+
+        return $generated;
+    }
+
     /**
-     * {@inheritdoc}
+     * @param Enumerator $enumerator
+     * @return string
+     * @throws \ReflectionException
+     * @throws \Throwable
      */
-    public function generateTraitForClass(ReflectionClass $class)
+    public function generateEnumeratorAccessors(Enumerator $enumerator)
     {
-        $code                  = '';
-        $add_collection_import = false;
+        $enum_class = $enumerator->getEnumeratorClass();
+        $reflector  = new \ReflectionClass($enum_class);
+        $class_name = $reflector->getShortName() . $this->enum_name_suffix;
+        $namespace  = $reflector->getNamespaceName() . '\\Generated';
+        $properties = [];
+
+        foreach ($reflector->getReflectionConstants() as $reflection_constant) {
+            $properties[] = new EnumItemInformation($reflection_constant);
+        }
+
+        return $this->enum_class->render([
+            'enum_class' => $enum_class,
+            'class_name' => $class_name,
+            'properties' => $properties,
+            'namespace'  => $namespace,
+            'username'   => get_current_user(),
+            'hostname'   => gethostname(),
```

Comment:
username and hostname cause 2 things and I'm not sure that's a good thing:
 - builds won't be reproducible
 - they will differ if you use a build server

They are not used in your template, so perhaps they can just be removed

Reply:
This is the way all other code is generated as well.

If there is a need for it, we can change it, but not part of this PR. 

I prefer to have everything consistent with the rest of this library rather than diverging from the way we're doing things.

Reply:
While I appreciate consistency, I still think they should be removed in this particular case.

Reply:
@hboomsma What do you think? Should be get rid of `get_current_user()` and `gethostname()` all together in this PR?

Reply:
@iltar @haroldiedema I think the removal of the `hostname` and `username` should be a separate concern, sho they should be added here.

Reply:
@hboomsma it's new code, which won't be used, so I think it shouldn't have to be added in the first place if it's already unused.

Reply:
@iltar the point is that all code generated by this library (_not just this_) is adding this comment with the hostname and username. Removing it here will only make the generation of this code inconsistent with the rest.

I'm leaving it here. No point in discussing further.